### PR TITLE
Fixed: Allow downloading any search result

### PR DESCRIPTION
**Pull request by ta264**
#### Database Migration
NO

#### Description
Allows you to download any search result in the interactive search dialog, even if the album or artist couldn't be parsed. Ones that couldn't be parsed get a red icon. 

#### Issues Fixed or Closed by this PR

* #441

### CODE REVIEW

**Comment by Qstick** [comment]
This was the case I was speaking of. Not sure it really hurts anything though as it will just fail out on import as it should.

image

File: src/NzbDrone.Core/DecisionEngine/DownloadDecisionMaker.cs [hunk 1]
```diff
@@ -94,10 +94,22 @@ private IEnumerable<DownloadDecision> GetAlbumDecisions(List<ReleaseInfo> report
                             if (remoteAlbum.Artist == null)
                             {
                                 decision = new DownloadDecision(remoteAlbum, new Rejection("Unknown Artist"));
+                                // shove in the searched artist in case of forced download in interactive search
+                                if (searchCriteria != null)
+                                {
+                                    remoteAlbum.Artist = searchCriteria.Artist;
+                                    remoteAlbum.Albums = searchCriteria.Albums;
+                                }
+                                remoteAlbum.DownloadAllowed = false;
```
Comment (Qstick) [inline src/NzbDrone.Core/DecisionEngine/DownloadDecisionMaker.cs]: Isn't `DownloadAllowed` always false right here? We shouldn't need this

File: src/NzbDrone.Core/DecisionEngine/DownloadDecisionMaker.cs [hunk 2]
```diff
@@ -94,10 +94,22 @@ private IEnumerable<DownloadDecision> GetAlbumDecisions(List<ReleaseInfo> report
...
                             }
                             else if (remoteAlbum.Albums.Empty())
                             {
                                 decision = new DownloadDecision(remoteAlbum, new Rejection("Unable to parse albums from release name"));
+                                if (searchCriteria != null)
+                                {
+                                    remoteAlbum.Albums = searchCriteria.Albums;
+                                }
+                                remoteAlbum.DownloadAllowed = false;
```
Comment (Qstick) [inline src/NzbDrone.Core/DecisionEngine/DownloadDecisionMaker.cs]: Isn't `DownloadAllowed` always false right here? We shouldn't need this

**Comment by Qstick** [comment]
Seems to work, comments above were mostly just cleanup and reference. Wonder if there is a way to flag these in activity queue?

**Comment by Qstick** [comment]
Looks good, have you tested what happens if a release gets pushed via API from outside of Lidarr and we can't parse it?  Do we still reject it given there is no search criteria to match up to?

**Comment by ta264** [comment]
If you do something like:
```
curl -H "X-Api-Key:xxx" X GET http://localhost:8686/lidarr/api/v1/release?albumId=752
curl -H "X-Api-Key:xxx" d "" -X POST "http://localhost:8686/lidarr/api/v1/release?guid=1_51a207c90ce17fb64fa43376f7a216a671759b78&indexerId=1
```
Then it will allow the download since the first `curl` triggered a search and we had a search criteria.  I've tested and the warning shows up.

It shouldn't affect anything coming through the RSS since there's no search criteria.  I haven't changed the release/push endpoint so I don't think it will affect that either.  In any case, these releases are all still "rejected" so any automatic handling shouldn't change.

**Pull request merged**